onchain/src/contracts/account.cairo

```diff
@@ -86,13 +93,49 @@ mod Account {
 
     #[constructor]
     fn constructor(
-        ref self: ContractState, public_key: felt252, approver: ContractAddress, limit: u256
+        ref self: ContractState, public_key: felt252, approver: ContractAddress, limit: u256,
     ) {
         self.account.initializer(:public_key);
         self.transaction_approval.initializer(:approver);
         self.weekly_limit.initializer(:limit);
     }
 
+    #[derive(Serde, Drop)]
+    struct Claim {
+        amount: u256,
+        nonce: felt252,
+        signature: Array<felt252>
+    }
+    #[starknet::interface]
```

Comment:
Could be better to have the interface outside of the contract, or even in a separate file

Reply:
yeah it should even be in a componentn but i counting on you for the refacto 😉